firmware/brake/kia_soul_ev_niro/tests/features/step_definitions/common.cpp

```diff
@@ -1,149 +1,11 @@
-extern uint8_t g_mock_mcp_can_send_msg_buf_len;
-extern uint8_t *g_mock_mcp_can_send_msg_buf_buf;
-
-extern unsigned short g_mock_dac_output_a;
-extern unsigned short g_mock_dac_output_b;
 
 extern volatile brake_control_state_s g_brake_control_state;
-extern volatile unsigned long g_mock_arduino_millis_return;
-
-// return to known state before every scenario
-BEFORE()
-{
-    g_mock_arduino_digital_write_count = 0;
-    memset(&g_mock_arduino_digital_write_pins, 0, sizeof(g_mock_arduino_digital_write_pins));
-    memset(&g_mock_arduino_digital_write_val, 0, sizeof(g_mock_arduino_digital_write_val));
-
-    memset(&g_mock_arduino_analog_read_return, 0, sizeof(g_mock_arduino_analog_read_return));
-
-    g_mock_mcp_can_check_receive_return = UINT8_MAX;
-    g_mock_mcp_can_read_msg_buf_id = UINT32_MAX;
-    memset(&g_mock_mcp_can_read_msg_buf_buf, 0, sizeof(g_mock_mcp_can_read_msg_buf_buf));
-
-    g_mock_mcp_can_send_msg_buf_id = UINT32_MAX;
-    g_mock_mcp_can_send_msg_buf_ext = UINT8_MAX;
-    g_mock_mcp_can_send_msg_buf_len = UINT8_MAX;
-
-    g_mock_dac_output_a = USHRT_MAX;
-    g_mock_dac_output_b = USHRT_MAX;
-
-    g_brake_control_state.enabled = false;
-    g_brake_control_state.operator_override = false;
-    g_brake_control_state.dtcs = 0;
-
-    // A small amount of time elapsed after boot
-    g_mock_arduino_millis_return = 1;
-}
-
-
-GIVEN("^brake control is enabled$")
-{
-    g_brake_control_state.enabled = 1;
-}
-
-
-GIVEN("^brake control is disabled$")
-{
-    g_brake_control_state.enabled = 0;
-}
-
-
-GIVEN("^the accelerator position sensors have a reading of (.*)$")
-{
-    REGEX_PARAM(int, sensor_val);
-
-    g_mock_arduino_analog_read_return[0] = sensor_val;
-}
-
-
-GIVEN("^the operator has applied (.*) to the accelerator$")
-{
-
-    REGEX_PARAM(int, brake_sensor_val);
-
-    g_mock_arduino_analog_read_return[0] = brake_sensor_val;
-
-    check_for_faults();
-}
-
-
-THEN("^control should be enabled$")
-{
-    assert_that(
-        g_brake_control_state.enabled,
-        is_equal_to(1));
-
-    assert_that(
-        g_mock_arduino_digital_write_pins[0],
-        is_equal_to(PIN_SPOOF_ENABLE));
-
-    assert_that(
-        g_mock_arduino_digital_write_val[0],
-        is_equal_to(HIGH));
-}
-
-
-THEN("^control should be disabled$")
-{
-    assert_that(
-        g_brake_control_state.enabled,
-        is_equal_to(0));
-
-    assert_that(
-        g_mock_arduino_digital_write_pins[0],
-        is_equal_to(PIN_SPOOF_ENABLE));
-
-    assert_that(
-        g_mock_arduino_digital_write_val[0],
-        is_equal_to(LOW));
-}
-
-
-THEN("^(.*) should be sent to DAC A$")
-{
-    REGEX_PARAM(int, dac_output_a);
-
-    assert_that(
-        g_mock_dac_output_a,
-        is_equal_to(dac_output_a));
-}
-
-
-THEN("^(.*) should be sent to DAC B$")
-{
-    REGEX_PARAM(int, dac_output_b);
 
-    assert_that(
-        g_mock_dac_output_b,
-        is_equal_to(dac_output_b));
-}
+#define FIRMWARE_NAME "brake"
```

Comment:
It is preferable to keep the following file contents order consistent unless there is some reason it won't work that way or there's a categorical ordering like in vehicle headers: `#include` -> `#define` -> `variable declarations` -> `variable definitions` -> `function declarations` -> `function definitions`

Reply:
I prefer that approach, and I'm on the fence about doing this. The trick is that the `FIRMWARE_NAME` define is being used in `../../common/testing/step_definitions/common.cpp`, so it needs to be defined before we include that file.

One option for this is to set this macro definition in the build system by adding

```cmake
add_definition(-DFIRMWARE_NAME=brake)
```

in some CMakeLists.txt where it's appropriate. Would you prefer this?

A slightly longer comment on this topic is that there are lot of structs and functions that do exactly the same thing but have duplicate names. This PR papers over these details by using some typedefs and defines to make things work, which is a trade off of reducing boiler plate and copy/paste with magic. This might have been a place where the magic got a bit too excessive, what are your thoughts?

Reply:
This issue has been addressed by refactoring the other files so that we can have a cleaner file structure.